src/titiler/extensions/titiler/extensions/render.py

```
@@ -0,0 +1,183 @@
+
+    params: RenderItem
+    links: List[Link]
+
+
+class RenderItemList(BaseModel):
+    """List of Render Items with links."""
+
+    renders: Dict[str, RenderItemWithLinks]
+    links: List[Link]
+
+
+@define
+class stacRenderExtension(FactoryExtension):
+    """Add /renders endpoint to a STAC TilerFactory."""
+
+    def register(self, factory: MultiBaseTilerFactory):
+        """Register endpoint to the tiler factory."""
+
+        def _validate_params(render: Dict) -> bool:
+            """Validate render related query params."""
+            # List of dependencies a `/tile` URL should validate
+            # Note: Those dependencies should only require Query() inputs
+            tile_dependencies = [
+                factory.reader_dependency,
+                factory.layer_dependency,
+                factory.dataset_dependency,
+                # Image rendering Dependencies
+                factory.rescale_dependency,
+                factory.color_formula_dependency,
+                factory.colormap_dependency,
+                factory.render_dependency,
+            ]
+
+            _values, errors = extract_query_params(tile_dependencies, render)
+
+            return errors
+
+        def _prepare_render_item(
+            render_id: str, render: Dict, request: Request, src_path: str
+        ) -> Dict:
+            """Prepare single render item."""
+            links = [
+                {
+                    "href": factory.url_for(
+                        request,
+                        "STAC Renders metadata",
+                        render_id=render_id,
+                    )
+                    + "?"
+                    + urlencode({"url": src_path}),
+                    "rel": "self",
+                    "type": "application/json",
+                    "title": f"STAC Renders metadata for {render_id}",
+                }
+            ]
+            errors = _validate_params(render)
+
+            if not errors:
```

`errors` here is a list of all validation errors for STAC render block. 
I decided to exclude links if there are errors, as I know they are invalid and will fail if you follow the link.
An alternative is to fail louder, e.g. with HTTP 400 and error details. Let me know if you prefer it.